pkg/controllermanager/controller/shoot/shoot_conditions_control.go

```diff
@@ -38,26 +38,30 @@ func (c *Controller) shootConditionsAdd(obj interface{}) {
 	c.shootConditionsQueue.Add(key)
 }
 
-func (c *Controller) filterSeedForShootConditions(obj, oldObj, _ client.Object, deleted bool) bool {
+func (c *Controller) filterSeedForShootConditions(obj, oldObj, _ client.Object, _ bool) bool {
```

Comment:
Such changes definitely deserve a unit test, as they are not particularly obvious to all developers. 
I appreciate that you added a comment explaining it. However, a unit test ensuring this logic would be even better.